ui/shared/ActionButton/ActionButton.tsx

```diff
@@ -0,0 +1,84 @@
+import { Button, Image, Text, useColorModeValue, chakra } from '@chakra-ui/react';
+import React from 'react';
+
+import type { AddressMetadataTagFormatted } from 'types/client/addressMetadata';
+
+import { route } from 'nextjs-routes';
+
+import config from 'configs/app';
+import * as mixpanel from 'lib/mixpanel/index';
+
+import LinkExternal from '../LinkExternal';
+
+type Props = {
+  data: NonNullable<AddressMetadataTagFormatted['meta']>;
+  className?: string;
+  txHash?: string;
+  source: 'Txn' | 'NFT collection' | 'NFT item';
+}
+
+const ActionButton = ({ data, className, txHash, source }: Props) => {
+  const defaultTextColor = useColorModeValue('blue.600', 'blue.300');
+  const defaultBg = useColorModeValue('gray.100', 'gray.700');
+
+  const { appID, textColor, bgColor, text, logoURL } = data;
+
+  const actionURL = data.actionURL?.replace('{chainId}', config.chain.id || '').replace('{txHash}', txHash || '');
+
+  const handleClick = React.useCallback(() => {
+    const info = appID || actionURL;
+    if (info) {
+      mixpanel.logEvent(mixpanel.EventTypes.PAGE_WIDGET, { Type: 'Action button', Info: info, Source: source });
+    }
+  }, [ source, appID, actionURL ]);
+
+  const content = (
+    <>
+      <Image
+        src={ logoURL }
+        alt={ `${ text } button` }
+        boxSize={ 5 }
+        borderRadius="sm"
+        mr={ 2 }
+      />
+      <Text fontSize="sm" fontWeight="500" color="currentColor">
+        { text }
+      </Text>
+    </>
+  );
+
+  return appID ? (
```

Comment:
I think here we should be more strict in displaying this component. What if I have a random public tag for a token address which has some sort of `actionURL` in metadata but it is not actually a marketplace or dapp URL? Should the button still be visible? Because we will already show a tag with this link in the page header.